src/Qbead.h

```diff
@@ -150,20 +257,20 @@ class Qbead {
   const uint8_t ix, iy, iz;
   const bool sx, sy, sz;
   float rbuffer[3];
-  float x, y, z, rx, ry, rz; // filtered and raw acc, in units of g
-  float t_acc, p_acc;        // theta and phi according to gravity
+  float rx, ry, rz;         // filtered and raw acc, in units of g
   float T_imu;             // last update from the IMU
 
-  float t_ble, p_ble; // theta and phi
+  State state;
```

Comment:
We do not need `state` to be member of `bead`. In particular, when we start looking into working with entangled states or mixed states, it would be much easier if they are decoupled

Reply:
Hello, thank you for your feedback. However, I dont fully understand, so the state object should not be part of the qbead object? I was thinking that each instance of qbead should include a state, that will manage all the state related functionality. As I understand now, state and qbead should be independent objects, and if a user wants to assign a state to a qbead, first a state obj should be created and then assigned to the qbead? but then both the qbead and the state will hold values of x,y,z,theta,phy, where the values associated to qbead are the once from imu or ble and the values of state are custom assigned?

Reply:
is it not possible to keep the state as an instance of qbead, and then if a user wants to assigned the values from the imu or ble to the state, do smth like this: eg.  bead.state.setXYZ(bead.x, bead.y, bead.x) ?

Reply:
Like (as an example IMU_reader) now:

void loop() {
  bead.readIMU();
  bead.clear();
  bead.setBloch_deg_smooth(bead.state.getTheta(), bead.state.getPhi(), color(255, 0, 255));
  bead.show();
  delay(5000);
}

be something like:

void loop() {
  bead.readIMU();
  bead.state.setXYZ(bead.x, bead.y, bead.z)
  bead.clear();
  bead.setBloch_deg_smooth(bead.state.getTheta(), bead.state.getPhi(), color(255, 0, 255));
  bead.show();
  delay(5000);
}

then all the imu and ble readouts are stored in qbead and assigned to state when needed?

Reply:
The issue is when we have more than one qbeads and want to have interesting entangled states.

Is the "state" simply the state of the LEDs? If that is the case, it really should not be a separate objects to begin with and having it tightly coupled as you do is indeed the most reasonable thing to do.

Is the "state" an object that represents a quantum state? Such an object can be plotted on the "output device" that is the LEDs of the qbead, but it is not the state of the LEDs itself. And we need a quantum state object in order to introduce the abstraction of gates and entanglement -- we do not need an LED state object. If we are going for this separation, then it makes more sense to be explicit, because it will not be always the case that there is one-to-one mapping of state-to-qbead -- we might want to have entangled states on multiple qbeads for instance.